### ISSUE TITLE
Functional or performance differences between using .UseLazyLoadingProxies() and Lazy loading without proxies

### ISSUE THREAD

**Issue by henrikdahl8240** [dotnet/efcore#24409]
https://docs.microsoft.com/en-us/ef/core/querying/related-data/lazy tells how to do lazy loading.

I tried using "Lazy loading without proxies" as it was introduced, I think with version 2 of EF Core. As far as I remember, Lazy loading without proxies was much slower than using .UseLazyLoadingProxies(), but I could be mistaken.

My question is, is there functional or performance differences between using the .UseLazyLoadingProxies() and Lazy loading without proxies?

Does using inheritance for entities introduce differences?

Basically, do you only win by not having to spend all the time and memory for the proxy objects made by .UseLazyLoadingProxies() if you use Lazy loading without proxies instead?

**Comment by ajcvickers** [comment]
@henrikdahl8240 Lazy loading proxies implement the "lazy loading without proxies" pattern. Using the pattern without the proxy overhead will not be slower.

**Comment by henrikdahl8240** [comment]
OK. If some classes will implement the "lazy loading without proxies" pattern explicitly, will the lazy loading proxies then just be created for the entities which do not implement the "lazy loading without proxies" pattern or it's a choice of the one approach exclusively or the other approach?

**Comment by ajcvickers** [comment]
Proxies is all or nothing.

**Comment by henrikdahl8240** [comment]
@ajcvickers OK. Do you have a suggestion for a snappy way of introducing the pattern to my entity model which consists of a couple of thousand classes or is it just the hard way?

**Comment by ajcvickers** [comment]
I know of no snappy way.

**Comment by henrikdahl8240** [comment]
@ajcvickers OK. Can the two approaches co-exist, because it takes a bit time to implement the pattern? I mean, if some of my classes implement the pattern and I use the .UseLazyLoadingProxies(), will the .UseLazyLoadingProxies() then just make proxy classes for all of my classes but it makes no harm if some entities implement the pattern by themselves?

**Comment by ajcvickers** [comment]
@henrikdahl8240 I don't know; that's not something I have tried.

**Comment by ajcvickers** [comment]
@henrikdahl8240 The initialization type is spent creating proxy types, so yes, you should see it improve to the same as it is when not using proxies. (It will also reduce memory usage significantly.)

**Comment by henrikdahl8240** [comment]
@ajcvickers Do you know, if it actually doesn't work at all, as it could look from the response from smitpatel at https://github.com/dotnet/efcore/issues/24453?
